cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -544,20 +592,14 @@ def _finish_method():
 
     instance.mgmt.refresh()
     prov_instance_raw = instance.mgmt.raw
-    try:
-        assert hasattr(prov_instance_raw, 'os-extended-volumes:volumes_attached')
-        volumes_attached = getattr(prov_instance_raw, 'os-extended-volumes:volumes_attached')
-        assert len(volumes_attached) == 1
-        volume_id = volumes_attached[0]["id"]
-        assert provider.mgmt.volume_exists(volume_id)
-        volume = provider.mgmt.get_volume(volume_id)
-        assert volume.size == 3
-    finally:
-        instance.cleanup_on_provider()
-        wait_for(lambda: not instance.exists_on_provider, num_sec=180, delay=5)
-        if "volume_id" in locals():  # To handle the case of 1st or 2nd assert
-            if provider.mgmt.volume_exists(volume_id):
-                provider.mgmt.delete_volume(volume_id)
+
+    assert hasattr(prov_instance_raw, 'os-extended-volumes:volumes_attached')
+    volumes_attached = getattr(prov_instance_raw, 'os-extended-volumes:volumes_attached')
+    assert len(volumes_attached) == 1
+    volume_id = volumes_attached[0]["id"]
```

Comment:
Optional: I don't see any need to create these variables if they are just used for the assertion.

Reply:
yeah, this looks odd. But I think this may be what the tests was actually intended to tested. I may try to make this more straightforward.

Reply:
On a second look I don't find these odd. They are just the checks that the test is probably intended to check.